Singletons/OnDemandHolderSingleton.java

```diff
@@ -0,0 +1,13 @@
+package Singletons;
+
+//this is the best solution
+//class creation in java is thread-safety, that means when class creates with an object, it is 100% thread-safe
+public class OnDemandHolderSingleton {
+    public static class SingletonHolder {
+        public static final SingletonHolder HOLDER_INSTANCE = new SingletonHolder();
```

Comment:
```suggestion
       static final SingletonHolder HOLDER_INSTANCE = new SingletonHolder();
```
interesting discussion about on demand holder idiom
https://stackoverflow.com/questions/21604243/correct-implementation-of-initialization-on-demand-holder-idiom